src/commands/deploy/deploy.ts

```diff
@@ -21,23 +21,26 @@ import { getRandomHexString } from "../../utils/randomUtils";
 import * as workspaceUtil from '../../utils/workspace';
 import { cancelWebsiteValidation, validateWebSite } from '../../validateWebSite';
 import { getWorkspaceSetting, updateWorkspaceSetting } from '../../vsCodeConfig/settings';
-import { getDefaultWebAppToDeploy } from '../getDefaultWebAppToDeploy';
 import { startStreamingLogs } from '../startStreamingLogs';
-import { IDeployWizardContext } from './IDeployWizardContext';
+import { getDefaultWebAppToDeploy } from './getDefaultWebAppToDeploy';
+import { IDeployWizardContext, WebAppSource } from './IDeployWizardContext';
 
 // tslint:disable-next-line:max-func-body-length cyclomatic-complexity
 export async function deploy(context: IActionContext, confirmDeployment: boolean, target?: vscode.Uri | SiteTreeItem | undefined): Promise<void> {
+    context.telemetry.properties.deployedWithDefaultApp = 'false';
 
-    let node: SiteTreeItem | undefined;
     const newNodes: SiteTreeItem[] = [];
-    context.telemetry.properties.deployedWithConfigs = 'false';
+    let node: SiteTreeItem | undefined;
     let deployFsPath: string | undefined;
+    let webAppSource: WebAppSource;
 
     if (target instanceof vscode.Uri) {
-        deployFsPath = target.fsPath;
         context.telemetry.properties.deploymentEntryPoint = 'fileExplorerContextMenu';
+        deployFsPath = target.fsPath;
+        webAppSource = WebAppSource.nodePicker;
```

Comment:
We don't actually know if this will be `nodePicker` until later on. I would prefer to delay setting this value until we actually know what it will be. So instead of this:
```
node = await getDefaultWebAppToDeploy(deployContext) || <SiteTreeItem>await ext.tree.showTreeItemPicker(WebAppTreeItem.contextValue, context);
```

It might look like this:
```
node = await getDefaultWebAppToDeploy(deployContext);
if (node) {
    webAppSource = WebAppSource.tree;
} else {
    webAppSource = WebAppSource.nodePicker;
    node = <SiteTreeItem>await ext.tree.showTreeItemPicker(WebAppTreeItem.contextValue, context);
}
```

Reply:
webAppSource gets overwritten in getDefaultWebApp to `WebAppSource.setting` if we used the setting, though this might not be good coding practice 🤔